src/xfer/cache.rs

```diff
@@ -25,38 +32,56 @@ pub enum CacheError {
 }
 
 #[derive(Debug)]
-pub struct Cache {
+pub struct Cache<State> {
     pub(crate) settings: CacheSettings,
+    state: PhantomData<State>,
+}
+
+impl<State> Cache<State> {
```

Comment:
Is there a reason you didn't do the more typical `CacheBuilder`/`Cache` split here?

Reply:
Actually, isn't the `CacheSettings` effectively the builder in this case anyway? Where would you prefer `Cache<CacheUninitialized>` over `CacheSettings`?

Reply:
1d5afed